hal/shared/platforms.h

```diff
@@ -39,5 +39,40 @@
 #define PLATFORM_B5SOM                      25
 #define PLATFORM_NEWHAL                     60000
 
+#if PLATFORM_ID == PLATFORM_GCC
+#define PRODUCT_SERIES                      "gcc"
+#endif
+
+#if PLATFORM_ID == PLATFORM_PHOTON_DEV || \
```

Comment:
We should probably not bother with anything other than `P1` and `PHOTON_PRODUCTION`.